command/013_config_upgrade.go

```diff
@@ -213,6 +213,11 @@ command and dealing with them before running this command again.
 		for _, rps := range file.RequiredProviders {
 			rewritePaths[path] = true
 			for _, rp := range rps.RequiredProviders {
+				// Skip internal providers
+				if rp.Type.IsBuiltIn() {
+					continue
+				}
```

Comment:
I think the consequence of skipping this is that we won't rewrite this attribute. That is, an input of:

```hcl
terraform {
  required_providers {
    terraform = "1.0.0"
  }
}
```

will result in the same output. Is that the intention? I'd expect either for the attribute to be dropped altogether or to be converted to the new object form with an explicit source of `"terraform.io/builtin/terraform"`.

Reply:
That's a good question! In my mind it makes more sense not to add a source block for internal providers; it's unnecessary and I think it might be more confusing to users. My preference is: if it's not required, don't add it.